Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            int numberOfIndividuals = PromptInt("Please enter the number of individuals you would like to calculate the BMI of: ");
+            while (numberOfIndividuals > 0)
+            {
+                PromptForPersonalData();
+
+                numberOfIndividuals--;
+            }
+        }
+
+        static void PromptForPersonalData()
+        {
+            string firstName = PromptString("Please enter the individual's first name: ");
+            string lastName = PromptString("Please enter the individual's last name: ");
+            int age = PromptInt("Please enter the individual's age: ");
+            float weight = PromptFloat("Please enter the individual's weight (in kg, example: 70.5): ");
+            float height = PromptFloat("Please enter the individual's height (in meters, example: 1.8): ");
+            float bodyMassIndex = CalculateBMI(weight, height);
+
+            Console.WriteLine($"{firstName} {lastName} is {age} years old, their weight is {weight} kg and their height is {height} m.");
+            Console.WriteLine($"{firstName} {lastName}'s BMI is: {bodyMassIndex}");
+        }
+
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateString(Console.ReadLine());
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateInt(Console.ReadLine());
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateFloat(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float height)
+        {
+            if (weight <= 0 || height <= 0)
```

Comment:
To a random reader this might pose a few questions: what does it mean?
``var isInvalidBmi = weight <= 0 || height <= 0;``
this, however, eliminates all the questions.
```cs
if(isInvalidBmi)
{
  ..
}
```

Reply:
Yes, you're exactly right. :) I've changed the code so now it validates the entered values (weight, height) in a separate function called IsValidBMI. It returns a bool and inside the CalculateBMI method, I only check for the returned value now and do the calculation if the validation succeeds. That should improve readability. One think I wasn't sure about is where to put the Console.WriteLine stuff but since the rest of the code had those in the validate functions, I follow the same pattern for BMI as well.

Reply:
for now it's okay, because those writeline thingies are more for logging rather than display (UI). Normally, you have a separation between the two.

Reply:
About that later in the lessons, so don't bother yourself with it yet ;)